screwdriver/api/api.go

```diff
@@ -33,102 +38,220 @@ type ResponseError struct {
-	body, err := ioutil.ReadAll(res.Body)
-	if err != nil {
-		return nil, fmt.Errorf("Reading response Body from Screwdriver API: %v", err)
-	}
-
-	switch res.StatusCode / 100 {
+func handleResponse(bodyBytes []byte, statusCode int) ([]byte, error) {
+	switch statusCode / 100 {
 	case 2:
-		return body, nil
+		res := new(util.CommandSpec)
+		err := json.Unmarshal(bodyBytes, res)
+		if err != nil {
+			return nil, fmt.Errorf("Screwdriver API Response unparseable: status=%d, err=%v", statusCode, err)
+		}
+		return bodyBytes, nil
 	case 4:
 		res := new(ResponseError)
-		err = json.Unmarshal(body, res)
+		err := json.Unmarshal(bodyBytes, res)
 		if err != nil {
-			return nil, fmt.Errorf("Unparseable error response from Screwdriver API: %v", err)
+			return nil, fmt.Errorf("Screwdriver API Response unparseable: status=%d, err=%v", statusCode, err)
 		}
 		return nil, res
 	case 5:
-		return nil, fmt.Errorf("%v: Screwdriver API has internal server error", res.StatusCode)
+		return nil, fmt.Errorf("Screwdriver API has internal server error: statusCode=%d", statusCode)
 	default:
-		return nil, fmt.Errorf("Unknown error happen while communicate with Screwdriver API")
+		return nil, fmt.Errorf("Unknown error happen while communicate with Screwdriver API: Statuscode=%d", statusCode)
 	}
 }
 
 // GetCommand returns Command from Screwdriver API
-func (c client) GetCommand(smallSpec *Command) (*Command, error) {
-	cmd := new(Command)
+func (c client) GetCommand(smallSpec *util.CommandSpec) (*util.CommandSpec, error) {
 	uri, err := url.Parse(c.baseURL)
 	if err != nil {
-		return nil, fmt.Errorf("The base Screwdriver API is invalid %q", c.baseURL)
+		return nil, fmt.Errorf("Failed to parse URL on GET: %v", err)
 	}
 	uri.Path = path.Join(uri.Path, "commands", smallSpec.Namespace, smallSpec.Name, smallSpec.Version)
-	req, err := http.NewRequest("GET", uri.String(), strings.NewReader(""))
+
+	// Request to api
+	const contentType = "application/json"
+	// No payload
+	payload := new(bytes.Buffer)
+
+	responseSpec, err := c.httpRequest("GET", uri.String(), contentType, payload)
 	if err != nil {
-		return nil, fmt.Errorf("Failed to create request about command to Screwdriver API: %v", err)
+		return nil, err
+	}
+
+	return responseSpec, nil
+}
+
+func specToPayloadBytes(commandSpec *util.CommandSpec) (specBytes []byte, err error) {
+	// Generate yaml payload bytes
+	commandSpecBytes, err := json.Marshal(commandSpec)
+	if err != nil {
+		return nil, err
+	}
+	var payload util.PayloadYaml
+	payload.Yaml = string(commandSpecBytes)
+	specBytes, err = json.Marshal(payload)
+	if err != nil {
+		return nil, err
+	}
+	return
+}
+
+func writeMultipartYaml(writer *multipart.Writer, specPath string, commandSpec *util.CommandSpec) error {
+	// Convert command spec to payload
+	specBytes, err := specToPayloadBytes(commandSpec)
+	if err != nil {
+		return fmt.Errorf("Failed to convert spec to bytes:%v", err)
+	}
+
+	// Set yaml filename
```

Comment:
I think this kind of obvious code comments are not needed.

As an other example httpRequest function do not need any comments I think.
https://github.com/screwdriver-cd/sd-cmd/pull/13/files#diff-cd10e99cd757d4857182f6d4e88c294cR163

Reply:
I agree you. I removed some comments that seems to be redundant. Thank you!